pinot-server/src/main/java/com/linkedin/pinot/server/starter/helix/SegmentLocks.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.server.starter.helix;
+
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+
+public class SegmentLocks {
+  private static final int NUM_LOCKS = 10000;
```

Comment:
this should be made configurable, since we allocate all the locks all the time. Or, maybe allocate locks on demand?

Reply:
There will be only 10K locks globally, which consumes less than 1MB memory.